source/slang/slang-emit-c-like.cpp

```diff
@@ -884,14 +883,32 @@ void CLikeSourceEmitter::emitSimpleValueImpl(IRInst* inst)
                     // This little hack is needed for gcc that if we have the expression
                     // int(-0x80000000) we get the warning: warning :  integer overflow in expression [-Woverflow]
                     // 0x80000000 and -0x80000000 mean the same thing when casted to 32 bit int, so we just flip the value here.
-                    IRIntegerValue value = litInst->value.intVal;
-                    value = (value == -0x80000000ll) ? -value : value;
-
+                    IRIntegerValue value = (litInst->value.intVal & 0xFFFFFFFF);
                     m_writer->emit("int(");
-                    m_writer->emit(value);
+                    m_writer->emit(int32_t(value));
                     m_writer->emit(")");
                     return;
                 }
+                case BaseType::Int8:
+                    {
+                        IRIntegerValue value = (litInst->value.intVal & 0xFF);
+                        value = (value == -0x80ll) ? -value : value;
```

Comment:
Why does Int8/Int16 have the ```value = (value == -0x80ll) ? -value : value;``` and the Int32 version removes it.

From the comment above it seems it's purpose is to avoid issues in gcc around int(-0x80000000). 

It's if the sequence that is desired (for say int8_t) is

```
// Does mask/sign extension
IRIntegerValue value = (int8_t)litInst->value.intVal;
// Makes sure MIN_INT8 is turned into unsigned form (to remove warning from gcc)
value = (value == -0x80ll) ? -value : value;
m_writer->emit("int8_t(");
m_writer->emit(value);
m_writer->emit(")");
```

Reply:
Yeah, I think that using the mask alone is sufficient to cover this case, so we should just remove all this negating logic and just use masking.

Reply:
The hack was there because there was a gcc version on a system we tested that failed to compile int(-0x80000000) in the way discussed. So that is a thing. That said I did try this out on a bunch of compilers/options on godbolt, and didn't repeat it.

So I guess it's ok, but it's possible it will arise again as an issue. On balance I'd leave it, but not  a show stopper.